decorate_test.go

```diff
@@ -313,6 +313,39 @@ func TestDecorateSuccess(t *testing.T) {
 		)
 		defer app.RequireStart().RequireStop()
 	})
+
+	t.Run("decoration must execute when required by a member of group", func(t *testing.T) {
+		type Drinks interface {
+		}
+		type Coffee struct {
+			Type  string
+			Name  string
+			Price int
+		}
+		type PriceService struct {
+			DefaultPrice int
+		}
+		app := fxtest.New(t,
```

Comment:
This test would pass even if none of the functions executed. What guarantees the assertions on 337 and 344 are called?

Reply:
Invokes are guaranteed to run (and the fact that it ran is guaranteed by `fx.RequireStart().RequireStop()` - if any Invokes fail to run, that fails the test). Invoke running successfully means assertion in line 344 are called.

And yes, line 337 in itself does not guarantee the execution, but the assertion in line 344 means both constructors `Provide`d here ran successfully.